packages/ts/form/src/Field.ts

```diff
@@ -255,7 +292,9 @@ export const field = directive(
         this.fieldState = fieldState;
 
         const updateValueFromElement = () => {
-          if (fieldState.strategy.checkValidity()) {
+          fieldState.strategy.checkValidity();
+          // When bad input is detected, skip reading new value in binder state
+          if (!fieldState.strategy.validity.badInput) {
```

Comment:
Just to understand what happens here: if in a date picker I change the initial value of `21/04/2023` to `21/04/2023a`, that value is not stored in the binder and the original one is safe, but if I change it to `21/04/202`, it is stored, right?

Reply:
Yes, assuming that `21/04/202` is a valid format for the `<vaadin-date-picker>`.

Reply:
Another easy way to break `<vaadin-date-picker>` parser that is even easier to reproduce is to enter too many digits, like `21/04/20231111`.